### PR TITLE
Feature/add tenants for get companies on prem

### PR DESCRIPTION
**Pull request by JWK**
Enhanced the GetAllCompaniesOfEnvironment procedure to differentiate between SaaS and on-premises environments.
Updated the base URL construction for on-premises environments to include the tenant name as a query parameter.
Ensured that the HTTP request method remains consistent for both environment types.

### CODE REVIEW

**Comment by Bertverbeek4PS** [comment]
@JWK thanks for your PR.
Could you also change the tooltip on page 50102 "ENVHUB Setup"
Specifies the redirect base URL api of the application that will be used to the Business Central integration. Like https://nav.contoso.com:7048/bc/api/v2.0/v1.0/companies.

to

Specifies the redirect base URL api of the application that will be used to the Business Central integration. Like https://nav.contoso.com:7048/bc

File: src/Communication.Codeunit.al [hunk 1]
```diff
@@ -45,12 +45,14 @@ codeunit 50103 "ENVHUB Communication"
...
     begin
         if ENVHUBSetup.Get() then;
-        if EnvironmentInformation.IsSaaS() then
-            exit(ENVHUBHttp.RequestMessage(StrSubstNo(UrlLbl, ENVHUBSetup."Tenant ID", environmentName), Method::Get, ''))
-        else
-            exit(ENVHUBHttp.RequestMessage(StrSubstNo(ENVHUBSetup."Base URL"), Method::Get, '')); //Change of baseUrl
+        if EnvironmentInformation.IsSaaS() then begin
+            exit(ENVHUBHttp.RequestMessage(StrSubstNo(saasUrlLbl, ENVHUBSetup."Tenant ID", environmentName), Method::Get, ''))
+        end else begin
+            exit(ENVHUBHttp.RequestMessage(StrSubstNo(onPremUrlLbl, ENVHUBSetup."Base URL", environmentName), Method::Get, '')); //Change of baseUrl with tenant/environment name
```
Comment (Bertverbeek4PS) [inline src/Communication.Codeunit.al]: Please remove the comment

**Comment by JWK** [comment]
> @JWK thanks for your PR. Could you also change the tooltip on page 50102 "ENVHUB Setup" Specifies the redirect base URL api of the application that will be used to the Business Central integration. Like https://nav.contoso.com:7048/bc/api/v2.0/v1.0/companies.
> 
> to
> 
> Specifies the redirect base URL api of the application that will be used to the Business Central integration. Like https://nav.contoso.com:7048/bc

I’ve updated the tooltip.

File: src/Communication.Codeunit.al (hunk 1 above)
Comment (Bertverbeek4PS) [reply src/Communication.Codeunit.al]: @JWK I meant the comment on line 55.
In the labels there must always be a comment.

File: src/Communication.Codeunit.al [hunk 2]
```diff
@@ -45,12 +45,13 @@ codeunit 50103 "ENVHUB Communication"
         ENVHUBHttp: Codeunit "ENVHUB Http";
         EnvironmentInformation: Codeunit "Environment Information";
         Method: Option Get,Post,Patch;
-        UrlLbl: Label 'https://api.businesscentral.dynamics.com/v2.0/%1/%2/api/v1.0/companies', comment = '%1 = Tenant ID, %2 = Environment name';
+        saasUrlLbl: Label 'https://api.businesscentral.dynamics.com/v2.0/%1/%2/api/v1.0/companies';
```
Comment (Bertverbeek4PS) [inline src/Communication.Codeunit.al]: Please add the comment here again

File: src/Communication.Codeunit.al [hunk 3]
```diff
@@ -45,12 +45,13 @@ codeunit 50103 "ENVHUB Communication"
...
+        onPremUrlLbl: Label '%1/api/v1.0/companies?tenant=%2';
```
Comment (Bertverbeek4PS) [inline src/Communication.Codeunit.al]: Please add the comment here again

**Review by Bertverbeek4PS — CHANGES_REQUESTED** [review]

File: src/Communication.Codeunit.al (hunk 2 above)
Comment (JWK) [reply src/Communication.Codeunit.al]: Added the following comment for SaasUrlLbl  '%1 = Tenant ID, %2 = Environment name';

File: src/Communication.Codeunit.al (hunk 3 above)
Comment (JWK) [reply src/Communication.Codeunit.al]: Added the following comment for OnPremUrlLbl '%1 = Base URL, %2 = Environment name';

File: src/Communication.Codeunit.al (hunk 1 above)
Comment (JWK) [reply src/Communication.Codeunit.al]: Removed comment on line 55 and added the label comments.

**Review by Bertverbeek4PS — APPROVED** [review]

**Pull request merged**